addons/api/addon-test-support/handlers/cache-daemon-search.js

```diff
@@ -69,16 +70,18 @@ export default function setupStubs(hooks) {
           .withArgs('searchCacheDaemon')
           .onCall(i)
           .returns({
-            [resourceName]: models.map((model) => {
-              // Use internal serializer to serialize the model correctly
-              // according to our mirage serializers
-              const modelData =
-                this.server.serializerOrRegistry.serialize(model);
+            [underscore(resourceNames[singularize(resourceName)])]: models.map(
```

Comment:
none blocking question: What is the reason behind adding underscore? Did we get case sensitive issues? or is for visual/design purposes?

BTW I didn't know this was available through `@ember/string`

Reply:
So the main issue is the change [here](https://github.com/hashicorp/boundary-ui/pull/2475/files#diff-b16b26b10ee0d49fd5b21a15ae0557b83b56f724edf52ab924b41737972ad124L144). Previously I was just array destructuring the value out of the return. Unfortunately now that they added more fields to the object, I can't guarantee getting the correct field so I had to grab it by key. This complicated things as this didn't accurately map the field in mirage to the actual response type now for some fields. 

The issue is for multiple word types. For example, we make the call for a resource type called `alias` as `store.query('alias')` which we map to `resolvable-aliases` for the daemon but the actual response field type is `resolvable_aliases`. This is further complicated by the usage of mirage where the resource types are plural and it's _`aliases`_ (instead of `alias`) which we need to return as `resolvable_aliases`.

Let me know if that's unclear, it's a bit convoluted but it's how our types are being mapped across our models both from API and mirage